test/test-MetadataStore.js

```diff
@@ -158,6 +158,22 @@ exports.test_async_insert_all = function*(assert) {
   assert.deepEqual(items[7], [3, 5]);
 };
 
+exports.test_async_get_by_cache_key = function*(assert) {
```

Comment:
nit: I'd add a couple more tests, if only for regression testing and ensuring the code path has been run:
1. testing for a key that doesn't exist
2. testing for an image type that doesn't exist